src/nunit.analyzers/ValuesUsage/ValuesUsageAnalyzer.cs

```diff
@@ -0,0 +1,96 @@
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using NUnit.Analyzers.Constants;
+using NUnit.Analyzers.Extensions;
+
+namespace NUnit.Analyzers.ValuesUsage
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class ValuesUsageAnalyzer : DiagnosticAnalyzer
+    {
+        private static readonly DiagnosticDescriptor parameterTypeMismatch = DiagnosticDescriptorCreator.Create(id: AnalyzerIdentifiers.ValuesParameterTypeMismatchUsage,
+                                                                                                                title: ValuesUsageAnalyzerConstants.ParameterTypeMismatchTitle,
+                                                                                                                messageFormat: ValuesUsageAnalyzerConstants.ParameterTypeMismatchMessage,
+                                                                                                                category: Categories.Structure,
+                                                                                                                defaultSeverity: DiagnosticSeverity.Error,
+                                                                                                                description: ValuesUsageAnalyzerConstants.ParameterTypeMismatchDescription);
```

Comment:
I'm often working on a small laptop, so cannot see all the code without scrolling

```suggestion
        private static readonly DiagnosticDescriptor parameterTypeMismatch = DiagnosticDescriptorCreator.Create(
            id: AnalyzerIdentifiers.ValuesParameterTypeMismatchUsage,
            title: ValuesUsageAnalyzerConstants.ParameterTypeMismatchTitle,
            messageFormat: ValuesUsageAnalyzerConstants.ParameterTypeMismatchMessage,
            category: Categories.Structure,
            defaultSeverity: DiagnosticSeverity.Error,
            description: ValuesUsageAnalyzerConstants.ParameterTypeMismatchDescription);
```